homeassistant/components/tplink/binary_sensor.py

```diff
@@ -0,0 +1,96 @@
+"""Support for TPLink binary sensors."""
+
+from __future__ import annotations
+
+from dataclasses import dataclass
+from typing import Final
+
+from kasa import Feature
+
+from homeassistant.components.binary_sensor import (
+    BinarySensorDeviceClass,
+    BinarySensorEntity,
+    BinarySensorEntityDescription,
+)
+from homeassistant.core import HomeAssistant, callback
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+
+from . import TPLinkConfigEntry
+from .entity import CoordinatedTPLinkFeatureEntity, TPLinkFeatureEntityDescription
+
+
+@dataclass(frozen=True, kw_only=True)
+class TPLinkBinarySensorEntityDescription(
+    BinarySensorEntityDescription, TPLinkFeatureEntityDescription
+):
+    """Base class for a TPLink feature based sensor entity description."""
+
+
+BINARYSENSOR_DESCRIPTIONS: Final = (
```

Comment:
```suggestion
BINARY_SENSOR_DESCRIPTIONS: Final = (
```

Reply:
Done in https://github.com/home-assistant/core/pull/120404